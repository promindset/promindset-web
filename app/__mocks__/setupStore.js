/* eslint-disable prettier/prettier */
/* eslint-disable prefer-template */
import { applyMiddleware } from 'redux'
import isPlainObject from 'lodash.isplainobject'
import { createBrowserHistory } from 'history'
import { routerMiddleware } from 'connected-react-router'
import { loggingMiddleware } from 'utils/loggingMiddleware'

const isFunction = arg => typeof arg === 'function'

const baseSetupStore = middlewares => getState =>
  applyMiddleware(...middlewares)(() => {
    let actions = []
    let listeners = []

    const self = {
      getState() {
        return isFunction(getState) ? getState(actions) : getState
      },

      getActions() {
        return actions
      },

      dispatch(action) {
        if (!isPlainObject(action)) {
          throw new Error(
            'Actions must be plain objects. ' +
              'Use custom middleware for async actions.'
          )
        }

        if (typeof action.type === 'undefined') {
          throw new Error(
            'Actions may not have an undefined "type" property. ' +
              'Have you misspelled a constant? ' +
              'Action: ' +
              JSON.stringify(action)
          )
        }

        actions.push(action)

        for (let i = 0; i < listeners.length; i++) {
          listeners[i]()
        }

        return action
      },

      clearActions() {
        actions = []
      },

      subscribe(cb) {
        if (isFunction(cb)) {
          listeners.push(cb)
        }

        return () => {
          const index = listeners.indexOf(cb)

          if (index < 0) {
            return
          }
          listeners.splice(index, 1)
        }
      },

      replaceReducer(nextReducer) {
        if (!isFunction(nextReducer)) {
          throw new Error('Expected the nextReducer to be a function.')
        }
      }
    }
    self.injectedReducers = {}
    return self
  })

export const setupStore = baseSetupStore([
  loggingMiddleware,
  routerMiddleware(createBrowserHistory())
])({})
