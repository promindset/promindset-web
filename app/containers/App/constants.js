export const INC_COUNTER = 'INCREASE_COUNTER'
export const DEC_COUNTER = 'DECREASE_COUNTER'
