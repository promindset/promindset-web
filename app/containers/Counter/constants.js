import { ActionNameFactory } from 'utils/actionFactory'

const getActionName = ActionNameFactory('containers', 'counter')

export const INC_COUNTER = getActionName('INCREASE_COUNTER')
export const DEC_COUNTER = getActionName('DECREASE_COUNTER')
