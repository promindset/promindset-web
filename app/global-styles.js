import { createGlobalStyle } from 'styled-components'

export const GlobalStyle = createGlobalStyle`
  html,
  body {
    background-color: #fafafa;
    font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  }
`
