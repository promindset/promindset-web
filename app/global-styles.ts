import { createGlobalStyle } from 'styled-components'

export const GlobalStyle = createGlobalStyle`
  html,
  body {
    background-color: #fafafa;
    font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  }

  body.fontLoaded {
    font-family: 'Open Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  }
`
