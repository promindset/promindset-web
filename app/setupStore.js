import { applyMiddleware, combineReducers, compose, createStore } from 'redux'
import { loggingMiddleware } from 'utils/loggingMiddleware'
import { createBrowserHistory } from 'history'
import { connectRouter, routerMiddleware } from 'connected-react-router'

const history = createBrowserHistory()

export const createReducer = (injectedReducers = {}) => {
  // base of all reducers which allows us to instantiate reducers
  // or injects them by invoking this function with the required reducer.
  const rootReducer = combineReducers({
    router: connectRouter(history),
    ...injectedReducers
  })
  return rootReducer
}

export const setupStore = (initialState = {}) => {
  const composeEnhancers =
    (process.env.NODE_ENV != 'production' &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})) ||
    compose

  // Enhancers are some middleware functions for our store such as logging
  // and then we chain required middleware with our compose function.
  const enhancers = applyMiddleware(
    loggingMiddleware,
    routerMiddleware(history)
  )

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(enhancers)
  )

  // To avoid a big mess setting up reducers around we have an injectReducer hook
  // which allows us to inject reducers into the store from within the function itself
  // by using ReactReduxContext instead of importing every reducer in the app.
  store.injectedReducers = {}

  // module.hot is an object provided from webpack-hot-middleware to allow hot reloading
  // our application preserving the state as well (only in development mode)
  if (module.hot) {
    module.hot.accept(createReducer, () => {
      store.replaceReducer(createReducer(store.injectedReducers))
    })
  }

  return store
}

export const store = (initialState = {}) => setupStore(initialState)
