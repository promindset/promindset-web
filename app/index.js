import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { createBrowserHistory } from 'history'
import { ConnectedRouter } from 'connected-react-router'
import { AppContainer } from 'containers/App'
import { store } from './setupStore'

const mountNode = document.getElementById('app')

ReactDOM.render(
  <Provider store={store}>
    <ConnectedRouter history={createBrowserHistory()}>
      <AppContainer />
    </ConnectedRouter>
  </Provider>,
  mountNode
)

if (module.hot) {
  module.hot.accept('containers/App', () => {
    ReactDOM.unmountComponentAtNode(mountNode)
    ReactDOM.render(
      <Provider store={store}>
        <ConnectedRouter history={createBrowserHistory()}>
          <AppContainer />
        </ConnectedRouter>
      </Provider>,
      mountNode
    )
  })
}
