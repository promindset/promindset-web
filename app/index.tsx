import React from 'react'
import ReactDOM from 'react-dom'

import { App } from './containers/App'

ReactDOM.render(
  <App compiler="TypeScript" framework="React" />,
  document.getElementById('app')
)
