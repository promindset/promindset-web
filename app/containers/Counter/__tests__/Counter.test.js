import React from 'react'
import { store } from 'app/setupStore'
import { Provider } from 'react-redux'
import TestRenderer from 'react-test-renderer'
import { CounterContainer } from '../index'
import { INC_COUNTER, DEC_COUNTER } from '../constants'

describe('Counter Container', () => {
  let component

  beforeEach(() => {
    store.dispatch = jest.fn()
    component = TestRenderer.create(
      <Provider store={store}>
        <CounterContainer />
      </Provider>
    )
  })

  test('snapshot render', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should dispatch increment action on inc button click', () => {
    TestRenderer.act(() =>
      component.root
        .findAllByType('button')
        .find(x => x.props.children == ['INC NUM'])
        .props.onClick()
    )
    expect(store.dispatch).toHaveBeenCalledTimes(1)

    expect(store.dispatch).toHaveBeenCalledWith({
      type: INC_COUNTER,
      payload: { amount: 1 }
    })
  })

  test('should dispatch decrement action on dec button click', () => {
    TestRenderer.act(() =>
      component.root
        .findAllByType('button')
        .find(x => x.props.children == ['DEC NUM'])
        .props.onClick()
    )
    expect(store.dispatch).toHaveBeenCalledTimes(1)

    expect(store.dispatch).toHaveBeenCalledWith({
      type: DEC_COUNTER,
      payload: { amount: 1 }
    })
  })
})
