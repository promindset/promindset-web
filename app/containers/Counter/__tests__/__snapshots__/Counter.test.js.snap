// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Container snapshot render 1`] = `
Array [
  <hr
    style={
      Object {
        "margin": "auto",
        "width": "50%",
      }
    }
  />,
  <h2
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    0
  </h2>,
  <div
    style={
      Object {
        "margin": "auto",
        "width": "max-content",
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "margin": "25px",
        }
      }
    >
      DEC NUM
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "margin": "25px",
        }
      }
    >
      INC NUM
    </button>
  </div>,
]
`;
